### PR TITLE
test: hash consensus: add more frames tests

### PR DESCRIPTION
**Pull request by DiRaiks**
Tests HashConsensus:
- Add returning `fastLaneLengthSlots` to `getFrameConfig` method in hashConsensus contract
 - add more tests for work with frames
 - update access control tests
 - add help methods to `HashConsensusTimeTravellable`
 - fix code style with prettier
 - add `INITIAL_EPOCH` const to `deployHashConsensus`

image
image

### CODE REVIEW

**Review by TheDZhon — APPROVED** [review]
Thank you for the contribution :rocket:

File: test/0.8.9/oracle/hash-consensus-frames.js [hunk 1]
```diff
@@ -26,36 +30,109 @@ const computeIthFrameStartTime = (frameIndex, epochsPerFrame, initialEpoch) =>
...
+      it('should revert if epoch < config.initialEpoc', async () => {
+        await consensus.setTimeInEpochs(INITIAL_EPOCH - 1)
+
+        await assert.revertsWithCustomError(consensus.setFrameConfig(1, 50), 'InitialEpochIsYetToArrive()')
+      })
+
+      it('should emit FrameConfigSet & FastLaneConfigSet event', async () => {
+        const tx = await consensus.setFrameConfig(100, 50)
+
+        assertEvent(tx, 'FrameConfigSet')
```
Comment (skozin) [inline test/0.8.9/oracle/hash-consensus-frames.js:96]: need to test for event arguments here and later, like it's done e.g. here: https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/test/0.8.9/oracle/hash-consensus-members.test.js#L110

File: contracts/0.8.9/test_helpers/oracle/HashConsensusTimeTravellable.sol [hunk 2]
```diff
@@ -72,4 +72,12 @@ contract HashConsensusTimeTravellable is HashConsensus {
     function advanceTimeByEpochs(uint256 numEpochs) external {
         _time += SECONDS_PER_SLOT * SLOTS_PER_EPOCH * numEpochs;
     }
+
+    function getFrameConfigFastLaneLengthSlots() external view returns (uint256 fastLaneLengthSlots) {
```
Comment (skozin) [inline contracts/0.8.9/test_helpers/oracle/HashConsensusTimeTravellable.sol:76]: this setting needs to be added as a return parameter to the `getFrameConfig` function of the main `HashConsensus` contract, I forgot to do this. could you please implement this as part of this PR?

File: test/0.8.9/oracle/hash-consensus-frames.js [hunk 3]
```diff
@@ -187,33 +262,34 @@ contract('HashConsensus', ([admin, member1, member2]) => {
...
-      assert.equal(+frame.reportProcessingDeadlineSlot, computeEpochFirstSlot(11) - 1)
-
-      await consensus.setFrameConfig(4, 0)
-
-      const newFrame = await consensus.getCurrentFrame()
-      assert.equal(+newFrame.refSlot, computeEpochFirstSlot(10) - 1)
-      assert.equal(+newFrame.reportProcessingDeadlineSlot, computeEpochFirstSlot(14) - 1)
-    })
+    it(
+      `decreasing the frame size may advance the current reference slot, ` + `but at least by the new frame size`,
```
Comment (skozin) [inline test/0.8.9/oracle/hash-consensus-frames.js:266]: string concatenation is not needed here anymore after reformatting

File: test/0.8.9/oracle/hash-consensus-frames.js [hunk 4]
```diff
@@ -161,7 +236,7 @@ contract('HashConsensus', ([admin, member1, member2]) => {
     })
 
     it(`decreasing the frame size cannot decrease the current reference slot`, async () => {
-      assert.equal(+await consensus.getTime(), computeTimestampAtEpoch(1))
+      assert.equal(+(await consensus.getTime()), computeTimestampAtEpoch(1))
```
Comment (skozin) [inline test/0.8.9/oracle/hash-consensus-frames.js:239]: could we change prettier settings to not add parentheses around await in cases like this one? it's much less readable this way imo

Comment (skozin) [reply test/0.8.9/oracle/hash-consensus-frames.js:239]: not critical, a nitpick

Comment (Jeday) [reply test/0.8.9/oracle/hash-consensus-frames.js:239]: Faced same issues with prettier in different PR. My prettier is pulling config from root of the repo but a lot of other files are not formatted at all. In a different PR, we can update config and format all(or just currently relevant) files.

File: test/0.8.9/oracle/hash-consensus-frames.js (hunk 1 above)
Comment (DiRaiks) [reply test/0.8.9/oracle/hash-consensus-frames.js]: Added

File: contracts/0.8.9/test_helpers/oracle/HashConsensusTimeTravellable.sol (hunk 2 above)
Comment (DiRaiks) [reply contracts/0.8.9/test_helpers/oracle/HashConsensusTimeTravellable.sol]: Added to contract

File: test/0.8.9/oracle/hash-consensus-frames.js (hunk 3 above)
Comment (DiRaiks) [reply test/0.8.9/oracle/hash-consensus-frames.js]: fixed

File: test/0.8.9/oracle/hash-consensus-frames.js (hunk 4 above)
Comment (TheDZhon) [reply test/0.8.9/oracle/hash-consensus-frames.js:241]: let's do it later in a single batch I suggest

**Review by TheDZhon — APPROVED** [review]
Magnificient, thanks :+1:

**Pull request merged**